test/unit/typeahead_spec.js

```diff
@@ -368,6 +367,28 @@ describe('Typeahead', function() {
     });
   });
 
+  describe('when clearOnSelected flag is set to true', function() {
+    it('clears input when selected', function() {
+      var spy = jasmine.createSpy();
+      var view = new Typeahead({
+        input: this.$input,
+        clearOnSelected: true,
+        hint: true,
+        datasets: {}
+      });
+      view.dropdown.getDatumForCursor.and.returnValue(testDatum);
+
+
+      var $e = jasmine.createSpyObj('event', ['preventDefault']);
+      view.$input.on('autocomplete:selected', spy);
+      view.input.trigger('enterKeyed', $e);
+
+      expect(spy).toHaveBeenCalled();
```

Comment:
Can you test the values that the callback receives?

Reply:
That's already tested in other tests, this behaviour doesn't change that

Reply:
Why test that it has been called. This feature could mess with the parameters if we do not know the actual implementation.